app/commands/track/update_build_status.rb

```diff
@@ -38,22 +38,16 @@ def component_health_statuses = [
     analyzer_health,
     test_runner_health,
     representer_health,
-    syllabus_health,
+    track.course? ? syllabus_health : :exemplar,
     practice_exercises_health
   ].tally
 
   def volunteers
-    volunteer_user_ids = User::ReputationPeriod.where(
-      period: :forever,
-      about: :track,
-      category: :any,
-      track_id: track.id
-    ).select(:user_id)
-    volunteers = User.with_attached_avatar.where(id: volunteer_user_ids).order(reputation: :desc).take(12)
+    volunteers = AssembleContributors.({ track_slug: track.slug, page: 1 })
 
     {
-      num_volunteers: volunteer_user_ids.distinct.count,
-      users: SerializeAuthorOrContributors.(volunteers)
+      num_volunteers: volunteers.dig(:meta, :total_count),
+      users: volunteers[:results].take(NUM_VOLUNTEERS)
```

Comment:
While I like this idea, I don't think you want to do it. That AssembleContibutors thing is really slow due to the contextual data.

Instead, I'd go with your existing code, but replace the `where...select` with:
```
    User::ReputationPeriod::Search.(track_id: track.id)
```

which I think does what you want.

Reply:
But then I don't have the right, per-track reputation for volunteers.

Reply:
I think this is the going to be the most efficient way:

```
users = User::ReputationPeriod::Search.(track_id: track.id)
track_rep_mapping = User::ReputationToken.where(track_id: track.id).where(user: users).group(:user_id).sum(:value)
```

Reply:
I've changed the implementation. Let me know what you think